presto-raptor/src/main/java/com/facebook/presto/raptor/storage/OrcStorageManager.java

```diff
@@ -457,6 +485,130 @@ private ShardInfo createShardInfo(UUID shardUuid, OptionalInt bucketNumber, Path
         return shardDelta(shardUuid, Optional.of(shard));
     }
 
+    private Collection<Slice> writeDeltaDeleteFile(UUID oldShardUuid, Optional<UUID> oldDeltaDeleteShardUuid, long transactionId, Block blockToDelete, OptionalInt bucketNumber)
```

Comment:
Too much details in a manager class. Let's move the logic into the new rewriter.